index.js

```diff
@@ -49,32 +49,42 @@ function walk (newNode, oldNode) {
 // update the children of elements
 // (obj, obj) -> null
 function updateChildren (newNode, oldNode) {
-  if (!newNode.childNodes || !oldNode.childNodes) return
+  var oldChildren = oldNode.childNodes
+  var newChildren = newNode.childNodes
+  if (!newChildren || !oldChildren) return
 
-  var newLength = newNode.childNodes.length
-  var oldLength = oldNode.childNodes.length
-  var length = Math.max(oldLength, newLength)
+  var newIndex = 0
+  for (var oldIndex = 0; oldIndex < oldChildren.length; oldIndex++) {
```

Comment:
Given that `oldChildren.length` is constant, we can evade doing an extra lookup on every loop:
```js
for (var oldIndex = 0, len = oldChildren.length; oldIndex < len; oldIndex++) {
```